aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -1539,9 +1539,14 @@ __global__ void indexSelectLargeIndex(cuda::detail::TensorInfo<T, IndexType> dst
                                       int64_t srcSelectDimSize) {
   // We stride over the output including the indexed dimension
   // (totalSize), and calculate the destination index point based on that
-  for (IndexType linearIndex = blockIdx.x * blockDim.x + threadIdx.x;
-       linearIndex < totalSize;
-       linearIndex += gridDim.x * blockDim.x) {
+  constexpr bool kPack4 = (sizeof(T) == 2);
+  constexpr int kElemsPerVec = kPack4 ? 4 : 1;
+
+  for (IndexType vecLinear = (blockIdx.x * blockDim.x + threadIdx.x) * kElemsPerVec;
```

Comment:
Is there any risk of overflow (can `IndexType` be 64-bit wide here)? If so, then `blockIdx.x * blockdim.x + threadIdx.x` should probably be cast to `IndexType` as they are implicitly 32-bit here.